### DISCUSSION TITLE
Add a option to drawning

### DISCUSSION THREAD

**Discussion by kopeek** [godotengine/godot-proposals#10921] [2D]
### Describe the project you are working on

I'm was working on a project that used drawning, but I stopped

### Describe the problem or limitation you are having in your project

I could use Line2D but I didn't used, I wanted to use draw functions etc. and I wanted to draw something next frame, but It cleared the whole drawning

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I want to Godot have an option, to just draw something without calling queue_redraw(), that's it

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I can be passe like an argument (If I want to queue_redraw() or not)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

In my opinion it can be easly done

### Is there a reason why this should be core and not an add-on in the asset library?

This is about improving 2D games

**Comment by Calinou** [comment]
Moved to discussion, as there is no concrete technical implementation provided.

Can you describe in more detail what this would be used for? Why is calling `queue_redraw()` a problem, and what issues did you encounter when using the existing [custom 2D drawing functions](https://docs.godotengine.org/en/stable/tutorials/2d/custom_drawing_in_2d.html)?

**Reply by kopeek** [reply]
I can't draw something different in different time, without it refreshing and clearing the drawnings, that's a problem that I encounter, It should be an option If I want to refresh or not

**Reply by Calinou** [reply]
I believe you should be able to draw within a SubViewport whose **Clear Mode** property is set to **Never**.

**Reply by KoBeWi** [reply]
I assume your problem is that `_draw()` method can only be called after again by using `queue_redraw()`?
You can workaround this limit by using RenderingServer methods directly, e.g. instead of `draw_rect()` use `canvas_item_add_rect()`. You can call it outside `_draw()`, but keep in mind that redrawing the node (with `queue_redraw()` or visibility change) will still clear it.

**Reply by CarpenterBlue** [reply]
Maybe having `queue_redraw()` without clearing the screen could be handy.
Perhaps `queue_overdraw()`? Just `queue_draw()` would likely be confusing.

**Reply by kopeek** [reply]
It would be good
